### ISSUE TITLE
`Style/RedundantParentheses` causes a false positive when using a one-line pattern matching in method argument

### ISSUE THREAD

**Issue by sanfrecce-osaka** [rubocop/rubocop#14217]
`Style/RedundantParentheses` causes a false positive when using a one-line pattern matching in method argument. Also it will causes a syntax error after auto correct.

This problem occurred when I tried to write the following code using `rspec`.

```ruby
expect((bar in baz)).to be true
```

## Expected behavior

The following code should be passed:

```ruby
foo((bar in baz))
```

## Actual behavior

The following code isn't passed:

```ruby
foo((bar in baz))  # 👮‍♂️ C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a method argument.
    ^^^^^^^^^^^^
```

And after auto correct, I get a syntax error.

```ruby
foo(bar in baz)  # 👮‍♂️ F: Lint/Syntax: unexpected ')', expecting end-of-input
        ^^
```

## RuboCop version

```
$ bundle exec rubocop -V
1.75.7 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.44.1, analyzing as Ruby 3.4, running on ruby 3.4.3) +server [arm64-darwin24]
  - rubocop-performance 1.25.0
  - rubocop-rails 2.32.0
  - rubocop-rspec 3.6.0
  - rubocop-rspec_rails 2.31.0
  - rubocop-factory_bot 2.27.1
```
